tests/TestCase/Database/Type/DateTypeTest.php

```diff
@@ -203,6 +208,96 @@ public function marshalProvider(): array
         return $data;
     }
 
+    /**
+     * Data provider for marshalWithTimezone()
+     *
+     * @return array
+     */
+    public function marshalWithTimezoneProvider(): array
+    {
+        Configure::write('Error.ignoredDeprecationPaths', [
+            'src/I18n/Date.php',
+        ]);
+
+        $defaultTimezone = date_default_timezone_get();
+        date_default_timezone_set('Europe/Vienna');
+        $date = new ChronosDate('@1392387900');
+
+        $data = [
+            // invalid types.
+            [null, null],
+            [false, null],
+            [true, null],
+            ['', null],
+            ['derpy', null],
+            ['2013-nope!', null],
+            ['2014-02-14 13:14:15', null],
+
+            // valid string types
+            ['1392387900', $date],
+            [1392387900, $date],
+            ['2014-02-14', new ChronosDate('2014-02-14')],
```

Comment:
Should we have a test for the `DateTime` types a;s well?

Reply:
Yes, we should. I will add that

Reply:
In the `DateTimeTypeTest` we already had a check to make sure that `Time::now()` returns `Time::now()` but not in combination with a different timezone.
Had to do the same fix in there as well.